src/main/java/de/rabea/gui/GuiApp.java

```diff
@@ -18,30 +17,26 @@ public void displayGameOptions() {
         viewUpdater.showBoardSizeOptionsView(this);
     }
 
-    public void createBoard(String boardSize) {
+    public Board createBoard(String boardSize) {
         if (boardSize.equals("3x3")) {
-            board = new Board(3);
+            return new Board(3);
         } else {
-            board = new Board(4);
+            return new Board(4);
         }
-        prepareGameForPlaying();
     }
 
-    public void prepareGameForPlaying() {
-        ClickCarrier carrier = new ClickCarrier();
-        GuiPlayer guiPlayer = new GuiPlayer(Mark.X, carrier);
-        GuiPlayer guiOpponent = new GuiPlayer(Mark.O, carrier);
+    public void prepareGameForPlaying(String boardSize) {
+        Board board = createBoard(boardSize);
+        GuiPlayer guiPlayer = new GuiPlayer(Mark.X);
```

Comment:
I did it because I thought it makes it more obvious that I'm not using one of the other players but the GuiPlayer. Happy to change it though and just call them `player` and `opponent`.